### PR TITLE
feat: major refactor of histograms

### PR DESCRIPTION
**Pull request by cabreraalex**

### CODE REVIEW

**Comment by cabreraalex** [comment]
In addition to fixing how histograms are calculated I:
* refactored rest API endpoint names to be consistent/REST-y
* Moved API and class files around to better organize

File: frontend/src/api/metadata.ts [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+			c.columnType !== ZenoColumnType.OUTPUT
+	);
+	const res = await fetch("/api/histograms", {
+		method: "POST",
+		headers: {
+			"Content-Type": "application/json",
+		},
+		body: JSON.stringify(requestedHistograms),
+	}).then((res) => res.json());
+	const histograms = new InternMap<string, HistogramEntry[]>(
```
Comment (xnought) [inline frontend/src/api/metadata.ts:37]: Not a review point, but this InternMap things is cool

File: frontend/src/api/metadata.ts [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+		requestedHistograms.map((col, i) => [col, res[i]]),
+		columnHash
+	);
+	return histograms;
+}
+
+let histogramCountAborter = null;
```
Comment (xnought) [inline frontend/src/api/metadata.ts:44]: This is probably a me problem. I am thoroughly confused what this aborter system is doing.

File: frontend/src/metadata/cells/metadata-cells/ContinuousMetadataCell.svelte [hunk 3]
```diff
@@ -67,6 +63,6 @@
 	<VegaLite
 		bind:view
 		spec={continuousVegaSpec($metricRange)}
-		data={chartData}
+		data={{ table: histogram.map((h) => Object.assign({}, h)) }}
```
Comment (xnought) [inline frontend/src/metadata/cells/metadata-cells/ContinuousMetadataCell.svelte:66]: Is this a vega-lite hack so it removes the vega keys (so it actually reloads)? If not, what is this Object.assign jazz for?

File: zeno/data_pipeline/zeno_backend.py [hunk 4]
```diff
@@ -13,23 +13,16 @@
...
-    MetadataType,
-    MetricKey,
-    Report,
-    Slice,
-    TableRequest,
-    ZenoColumn,
-    ZenoColumnType,
-    ZenoState,
-)
+from zeno.classes.base import MetadataType, ZenoColumnType
```
Comment (xnought) [inline zeno/data_pipeline/zeno_backend.py:21]: These new imports are so satisfying to look at

File: frontend/src/api/metadata.ts (hunk 1 above)
Comment (cabreraalex) [reply frontend/src/api/metadata.ts:37]: Agreed :)

File: frontend/src/metadata/cells/metadata-cells/ContinuousMetadataCell.svelte (hunk 3 above)
Comment (cabreraalex) [reply frontend/src/metadata/cells/metadata-cells/ContinuousMetadataCell.svelte:66]: Exactly what it is, I had a nightmare trying to get this to work. Added a clarifying comment

**Pull request merged**